### ISSUE TITLE
Add live graphical updates for the training process

### ISSUE THREAD

**Issue by ymoisan** [NRCan/geo-deep-learning#8]
See #55.

It would be nice to have a live graphing system to visualize per class metrics as the training/validation process evolves.  We could also set alarms (and highlight in graph ?) when the loss function starts increasing and potentially stop the process.

We currently dump sklearn's [classification report](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) in log files.  We could try to plug that output in something like this : [python-live-graph-update-from-a-changing-text-file](https://stackoverflow.com/questions/33457040/python-live-graph-update-from-a-changing-text-file)

Longer term goal : implement structured logging.  That would allow us to analyze several log files (e.g. training a dataset with different parameters over time) in log munging software.  

Interesting item : [Eliot](https://github.com/ScatterHQ/eliot), the **causal** logging library. "Most logging systems tell you *what* happened in your application, whereas eliot also tells you *why* it happened."

**Comment by ymoisan** [comment]
See [Task manager project](https://github.com/lemairecarl/experiment-manager).
